common/channel/src/libra_channel.rs

```diff
@@ -62,10 +60,17 @@ impl<K: Eq + Hash + Clone, M> Sender<K, M> {
     }
 }
 
+impl<K: Eq + Hash + Clone, M> Clone for Sender<K, M> {
```

Comment:
does `#[derive(Clone)]` not work on `pub struct Sender`?

Reply:
Yeah, it doesn't. For a generic struct, `#[derive]` implements `Clone` conditionally by adding bound Clone on generic parameters.

In our case, we dont require that the generic type `M` implement `Clone`